packages/runtime/lib/worker/itc.js

```diff
@@ -0,0 +1,154 @@
+'use strict'
+
+const { once } = require('node:events')
+const { parentPort } = require('node:worker_threads')
+
+const { printSchema } = require('graphql')
+const { ITC } = require('@platformatic/itc')
+
+const errors = require('../errors')
+
+function setupITC (app, service, dispatcher) {
+  const itc = new ITC({ port: parentPort })
+
+  itc.handle('start', async () => {
+    const status = app.getStatus()
+
+    if (status === 'starting') {
+      await once(app, 'start')
+    } else {
+      await app.start()
+    }
+
+    if (service.entrypoint) {
+      await app.listen()
+    }
+
+    dispatcher.replaceServer(app.appConfig.useHttp ? app.server.url : app.server)
+
+    return service.entrypoint ? app.server.url : null
+  })
+
+  itc.handle('stop', async () => {
+    const status = app.getStatus()
+
+    if (status === 'starting') {
+      await once(app, 'start')
+    }
+
+    if (status !== 'stopped') {
+      await app.stop()
+    }
+
+    setImmediate(() => {
+      dispatcher.interceptor.close()
+      itc.close()
+    })
```

Comment:
why the setImmediate? Why don't we await those?

Reply:
While we could await the `dispatcher.interceptor.close()`, we still have to defer `itc.close` method to next iteration so that this method will finish and the ITC will respond to the other side.

Reply:
I'll modify the ITC to support a mechanism to close after a method finishes, so we don't need the `setImmediate`.